ebsco_adapter/ebsco_adapter_iceberg/src/steps/trigger.py

```diff
@@ -1,60 +1,167 @@
 import argparse
+import os
+import re
+import tempfile
+from datetime import datetime
 from typing import Any
 
+import boto3
 from pydantic import BaseModel
 
+from ebsco_ftp import EbscoFtp
 from steps.loader import EbscoAdapterLoaderEvent
+from utils.aws import get_ssm_parameter, list_s3_keys
+
+ssm_param_prefix = "/catalogue_pipeline/ebsco_adapter"
+s3_bucket = os.environ.get("S3_BUCKET", "wellcomecollection-platform-ebsco-adapter")
+s3_prefix = os.environ.get("S3_PREFIX", "dev")
+ftp_s3_prefix = os.path.join(s3_prefix, "ftp_v2")
 
 
 class EbscoAdapterTriggerConfig(BaseModel):
     is_local: bool = False
 
 
-class EbscoAdapterTriggerEvent(BaseModel):
-    job_id: str | None = None
+class EventBridgeTriggerEvent(BaseModel):
+    time: str
+
+
+def get_most_recent_valid_file(filenames: list[str]) -> str | None:
```

Comment:
Suggest this is a bit simpler: https://github.com/wellcomecollection/catalogue-pipeline/commit/147ca4681a2eee6b13abb2b15932441098784677

That change will fail if it encounters a badly formed date, but arguably that is a stopping error (rather than a silent ignore) as it's an unexpected thing to have happen.